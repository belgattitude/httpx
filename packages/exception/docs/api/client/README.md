[**@httpx/exception**](../README.md)

***

[@httpx/exception](../README.md) / client

# client

## Index

### Classes

- [HttpBadRequest](classes/HttpBadRequest.md)
- [HttpConflict](classes/HttpConflict.md)
- [HttpExpectationFailed](classes/HttpExpectationFailed.md)
- [HttpFailedDependency](classes/HttpFailedDependency.md)
- [HttpForbidden](classes/HttpForbidden.md)
- [HttpGone](classes/HttpGone.md)
- [HttpImATeapot](classes/HttpImATeapot.md)
- [HttpLengthRequired](classes/HttpLengthRequired.md)
- [HttpLocked](classes/HttpLocked.md)
- [HttpMethodNotAllowed](classes/HttpMethodNotAllowed.md)
- [HttpMisdirectedRequest](classes/HttpMisdirectedRequest.md)
- [HttpNotAcceptable](classes/HttpNotAcceptable.md)
- [HttpNotFound](classes/HttpNotFound.md)
- [HttpPayloadTooLarge](classes/HttpPayloadTooLarge.md)
- [HttpPaymentRequired](classes/HttpPaymentRequired.md)
- [HttpPreconditionFailed](classes/HttpPreconditionFailed.md)
- [HttpPreconditionRequired](classes/HttpPreconditionRequired.md)
- [HttpProxyAuthenticationRequired](classes/HttpProxyAuthenticationRequired.md)
- [HttpRangeNotSatisfiable](classes/HttpRangeNotSatisfiable.md)
- [HttpRequestHeaderFieldsTooLarge](classes/HttpRequestHeaderFieldsTooLarge.md)
- [HttpRequestTimeout](classes/HttpRequestTimeout.md)
- [HttpTooEarly](classes/HttpTooEarly.md)
- [HttpTooManyRequests](classes/HttpTooManyRequests.md)
- [HttpUnauthorized](classes/HttpUnauthorized.md)
- [HttpUnavailableForLegalReasons](classes/HttpUnavailableForLegalReasons.md)
- [HttpUnprocessableEntity](classes/HttpUnprocessableEntity.md)
- [HttpUnsupportedMediaType](classes/HttpUnsupportedMediaType.md)
- [HttpUpgradeRequired](classes/HttpUpgradeRequired.md)
- [HttpUriTooLong](classes/HttpUriTooLong.md)
