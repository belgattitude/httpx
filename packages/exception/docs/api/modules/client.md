[@httpx/exception](../README.md) / client

# Module: client

## Table of contents

### Classes

- [HttpBadRequest](../classes/client.HttpBadRequest.md)
- [HttpConflict](../classes/client.HttpConflict.md)
- [HttpExpectationFailed](../classes/client.HttpExpectationFailed.md)
- [HttpFailedDependency](../classes/client.HttpFailedDependency.md)
- [HttpForbidden](../classes/client.HttpForbidden.md)
- [HttpGone](../classes/client.HttpGone.md)
- [HttpImATeapot](../classes/client.HttpImATeapot.md)
- [HttpLengthRequired](../classes/client.HttpLengthRequired.md)
- [HttpLocked](../classes/client.HttpLocked.md)
- [HttpMethodNotAllowed](../classes/client.HttpMethodNotAllowed.md)
- [HttpMisdirectedRequest](../classes/client.HttpMisdirectedRequest.md)
- [HttpNotAcceptable](../classes/client.HttpNotAcceptable.md)
- [HttpNotFound](../classes/client.HttpNotFound.md)
- [HttpPayloadTooLarge](../classes/client.HttpPayloadTooLarge.md)
- [HttpPaymentRequired](../classes/client.HttpPaymentRequired.md)
- [HttpPreconditionFailed](../classes/client.HttpPreconditionFailed.md)
- [HttpPreconditionRequired](../classes/client.HttpPreconditionRequired.md)
- [HttpProxyAuthenticationRequired](../classes/client.HttpProxyAuthenticationRequired.md)
- [HttpRangeNotSatisfiable](../classes/client.HttpRangeNotSatisfiable.md)
- [HttpRequestHeaderFieldsTooLarge](../classes/client.HttpRequestHeaderFieldsTooLarge.md)
- [HttpRequestTimeout](../classes/client.HttpRequestTimeout.md)
- [HttpTooEarly](../classes/client.HttpTooEarly.md)
- [HttpTooManyRequests](../classes/client.HttpTooManyRequests.md)
- [HttpUnauthorized](../classes/client.HttpUnauthorized.md)
- [HttpUnavailableForLegalReasons](../classes/client.HttpUnavailableForLegalReasons.md)
- [HttpUnprocessableEntity](../classes/client.HttpUnprocessableEntity.md)
- [HttpUnsupportedMediaType](../classes/client.HttpUnsupportedMediaType.md)
- [HttpUpgradeRequired](../classes/client.HttpUpgradeRequired.md)
- [HttpUriTooLong](../classes/client.HttpUriTooLong.md)
