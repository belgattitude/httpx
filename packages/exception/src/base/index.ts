export { HttpException } from './HttpException';
export { HttpServerException } from './HttpServerException';
export { HttpClientException } from './HttpClientException';
