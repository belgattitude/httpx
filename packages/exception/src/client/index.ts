export { HttpBadRequest } from './HttpBadRequest';
export { HttpConflict } from './HttpConflict';
export { HttpExpectationFailed } from './HttpExpectationFailed';
export { HttpFailedDependency } from './HttpFailedDependency';
export { HttpForbidden } from './HttpForbidden';
export { HttpGone } from './HttpGone';
export { HttpImATeapot } from './HttpImATeapot';
export { HttpLengthRequired } from './HttpLengthRequired';
export { HttpLocked } from './HttpLocked';
export { HttpMisdirectedRequest } from './HttpMisdirectedRequest';
export { HttpMethodNotAllowed } from './HttpMethodNotAllowed';
export { HttpNotAcceptable } from './HttpNotAcceptable';
export { HttpNotFound } from './HttpNotFound';
export { HttpPayloadTooLarge } from './HttpPayloadTooLarge';
export { HttpPaymentRequired } from './HttpPaymentRequired';
export { HttpPreconditionFailed } from './HttpPreconditionFailed';
export { HttpPreconditionRequired } from './HttpPreconditionRequired';
export { HttpProxyAuthenticationRequired } from './HttpProxyAuthenticationRequired';
export { HttpRangeNotSatisfiable } from './HttpRangeNotSatisfiable';
export { HttpRequestHeaderFieldsTooLarge } from './HttpRequestHeaderFieldsTooLarge';
export { HttpRequestTimeout } from './HttpRequestTimeout';
export { HttpTooEarly } from './HttpTooEarly';
export { HttpTooManyRequests } from './HttpTooManyRequests';
export { HttpUnauthorized } from './HttpUnauthorized';
export { HttpUnavailableForLegalReasons } from './HttpUnavailableForLegalReasons';
export { HttpUnprocessableEntity } from './HttpUnprocessableEntity';
export { HttpUnsupportedMediaType } from './HttpUnsupportedMediaType';
export { HttpUpgradeRequired } from './HttpUpgradeRequired';
export { HttpUriTooLong } from './HttpUriTooLong';
