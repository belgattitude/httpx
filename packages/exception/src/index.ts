export type {
  HttpExceptionParams,
  HttpStatusCode,
  HttpExceptionParamsWithStatus,
} from './types';
export * from './base';
export * from './client';
export * from './server';
export * from './typeguards';
export * from './factory';
