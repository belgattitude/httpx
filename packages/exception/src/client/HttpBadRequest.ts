import { HttpClientException } from '../base';
import type { HttpExceptionParams } from '../types';
import { getSuper } from '../utils';

/**
 * 400 Bad Request (client)
 *
 * The server cannot or will not process the request due to something that is perceived to be a client error
 * (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).
 *
 * @see https://httpstatus.in/400/
 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
 */
export class HttpBadRequest extends HttpClientException {
  static readonly STATUS = 400;
  constructor(msgOrParams?: HttpExceptionParams | string) {
    const name = 'BadRequest';
    super(400, getSuper(name, msgOrParams));
    Object.setPrototypeOf(this, HttpBadRequest.prototype);
    this.name = `Http${name}`;
  }
}
