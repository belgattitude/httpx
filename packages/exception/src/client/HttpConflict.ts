import { HttpClientException } from '../base';
import type { HttpExceptionParams } from '../types';
import { getSuper } from '../utils';

/**
 * 409 Conflict (client)
 *
 * This response is sent when a request conflicts with the current state of the server.
 *
 * @see https://httpstatus.in/409/
 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409
 */
export class HttpConflict extends HttpClientException {
  static readonly STATUS = 409;
  constructor(msgOrParams?: HttpExceptionParams | string) {
    const name = 'Conflict';
    super(409, getSuper(name, msgOrParams));
    Object.setPrototypeOf(this, HttpConflict.prototype);
    this.name = `Http${name}`;
  }
}
