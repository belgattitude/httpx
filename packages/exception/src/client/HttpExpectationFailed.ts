import { HttpClientException } from '../base';
import type { HttpExceptionParams } from '../types';
import { getSuper } from '../utils';

/**
 * Client status 417
 *
 * The HTTP 417 Expectation Failed client error response code indicates that the expectation given
 * in the request's Expect header could not be met.
 *
 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/417
 * @see https://httpstatus.in/417/
 */
export class HttpExpectationFailed extends HttpClientException {
  static readonly STATUS = 417;
  constructor(msgOrParams?: HttpExceptionParams | string) {
    const name = 'ExpectationFailed';
    super(417, getSuper(name, msgOrParams));
    Object.setPrototypeOf(this, HttpExpectationFailed.prototype);
    this.name = `Http${name}`;
  }
}
