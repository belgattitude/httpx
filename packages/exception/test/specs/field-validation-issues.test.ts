import { HttpUnprocessableEntity } from '../../src/client';
import type { HttpValidationIssue } from '../../src/types/HttpValidationIssue';

describe('HttpUnprocessableEntity (422) with field validation issues', () => {
  const issues: HttpValidationIssue[] = [
    {
      message: 'Invalid email',
      path: 'email',
      code: 'invalid_email',
    },
    {
      message: 'Invalid address',
      path: ['addresses', 0, 'line1'],
      code: 'empty_string',
    },
  ];
  it('should return unmodified http validation issues', () => {
    const e422 = new HttpUnprocessableEntity({
      issues,
    });
    expect(e422.issues).toStrictEqual(issues);
    expect(e422).toMatchSnapshot();
  });
});
