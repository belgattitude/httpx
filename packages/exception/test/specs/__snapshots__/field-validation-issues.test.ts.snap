// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HttpUnprocessableEntity (422) with field validation issues > should return unmodified http validation issues 1`] = `[HttpUnprocessableEntity: Unprocessable entity]`;
